db/db.go

```diff
@@ -102,18 +102,18 @@ func get(ctx context.Context, db *sql.DB, query string, args ...interface{}) (st
 
 	buf := []byte{}
 	err := row.Scan(&buf)
-	if err == nil {
-		return string(buf), nil
-	}
-
-	if err != sql.ErrNoRows {
-		err = errors.Wrap(err, "SELECT")
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return "", errors.Wrap(func() error {
+				return func() error {
+					return err
+				}()
+			}(), "hardware not found")
```

Comment:
what is going on here? :D. Why isn't this just `return "", errors.Wrap(err, "hardware not found")`?

Reply:
This is implementing the method for the `Cause` to work, basically making the `sql.ErrNoRows` error the "cause"

Reply:
Should I just create a global var with the value `HardwareNotFound` and then check for that instead?